psignifit/_result.py

```diff
@@ -158,6 +158,35 @@ def slope(self, stimulus_level: np.ndarray, estimate_type: Optional[EstimateType
         sigmoid = self.configuration.make_sigmoid()
         return sigmoid.slope(stimulus_level, param['threshold'], param['width'], param['gamma'], param['lambda'])
 
+    def proportion_correct(self, stimulus_level: np.ndarray,  with_eta: bool = False,
+                           estimate_type: Optional[EstimateType]=None, random_state=None) -> np.ndarray:
+        """ Proportion correct corresponding to the given stimulus levels for the fitted psychometric function.
+
+        Args:
+            stimulus_level: stimulus levels.
+            with_eta: if set after computing proportion correct values adds noise
+                to the data so that its variance is compatible with the estimated
+                overdispersion parameter `eta`.
+            estimate_type: Type of estimate, either "MAP" or "mean".
+                If None, the value of `estimate_type` in `Result.configuration` is used instead.
+            random_state: Random state used to generate the additional variance in the data if with_eta is True.
+                If None, NumPy's default random number generator is used.
+        Returns:
+            Proportion correct values for the fitted psychometric function at the given stimulus levels.
+        """
+        stimulus_level, param = np.asarray(stimulus_level), self.get_parameter_estimate(estimate_type)
+        if with_eta:
+            out = psychometric_with_eta(stimulus_level, param['threshold'], param['width'],
+                                        param['gamma'], param['lambda'],
+                                        self.configuration.make_sigmoid(),
+                                        param['eta'], random_state=None)
+        else:
+            sigmoid = self.configuration.make_sigmoid()
```

Comment:
small refactoring: define `sigmoid` outside, use in both branches

Reply:
fixed in 9ebde5cd16a4175f9caa946dd02a3e51f303e8a5